### PR TITLE
Improve package nodes visibility

### PR DESCRIPTION
**Pull request by dnenov**
### Purpose

A visual update to the Node border to package nodes. In zoomed-out state, a purple border allows the user to easily distinguish between Dynamo core nodes and external, package nodes. The bottom-left corner of the Node is now tinted with the same color as the workspace background again allowing the user to quickly see which nodes come from external packages. Finally the package name, as filled-in by the package owner, is shown inside the node tooltip.

#### Screenshots of UI changes
_Border states_

border states

_Package names_

package name

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [x] This PR contains no files larger than 50 MB 

### Release Notes

- Implement UI changes as per [Figma link](https://www.figma.com/file/MZRyq8azmRaQgdRKUkS8qJ/Package-node-visual-indicator?type=design&node-id=75-1354&mode=design&t=r0eewqZpEChsdUgz-0) that showcases to users what nodes
come from a package.
- Ensure Zoom State hierarchy is correct and test all cases where three states may be present.
- Inject Package name into Description of each node.
- Ensure Python Script node that has Engine text doesn’t overlap with new triangle corner package signifier

### Reviewers

@reddyashish 
@mmisol 

### FYIs

@QilongTang

### CODE REVIEW

**Review by reddyashish — APPROVED** [review]

File: src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs [hunk 1]
```diff
@@ -1342,6 +1354,12 @@ internal SolidColorBrush GetBorderColor()
 
             ResetColorGlyphs();
 
+
+            if (this.NodeModel.IsCustomFunction)
+            {
+                result = (SolidColorBrush)SharedDictionaryManager.DynamoColorsAndBrushesDictionary["NodeCustomColor"];
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:1360]: can you use nameof?

Comment (mjkkirschner) [reply src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:1360]: perhaps do some null checking?

Comment (dnenov) [reply src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:1361]: > can you use nameof?

Hi @mjkkirschner ! I don't know if I can here :/ This is a dictionary key, and I am using similar way of getting the value to this, for example: https://github.com/DynamoDS/Dynamo/blob/ea6380044dd71f54c63256a601aeebcf8ba60183/src/VisualizationTests/HelixWatch3DViewModelTests.cs#L618

Check for nulls - done!!

**Review by reddyashish — APPROVED** [review]
Will merge once the job passes.

File: src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs [hunk 2]
```diff
@@ -241,6 +241,18 @@ public string OriginalName
             get { return nodeLogic.GetOriginalName(); }
         }
 
+        /// <summary>
+        /// Returns the Package name of the node, if it comes from a package
+        /// </summary>
+        [JsonProperty(Order = 9)]
+        public string PackageName
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:248]: why is this being serialized? It looks like you never read the serialized data?

Comment (dnenov) [reply src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:248]: You are right, I will just ignore it, apologies!

Comment (dnenov) [reply src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:248]: Big thanks for this spot check, @mjkkirschner , hopefully it's fixed with this [PR](https://github.com/DynamoDS/Dynamo/pull/14216)

File: test/core/dummy_node/2080_JSONTESTCRASH undo_redo.dyn [hunk 3]
```diff
@@ -82,23 +82,25 @@
...
         "IsSetAsInput": false,
         "IsSetAsOutput": false,
-        "Name": "Code Block",
-        "ShowGeometry": true,
         "Excluded": false,
+        "ShowGeometry": true,
         "X": 76.8,
-        "Y": 162.4
+        "Y": 162.4,
+        "PackageName": ""
```
[the remark this answers is not in the document]
Comment (dnenov) [reply test/core/dummy_node/2080_JSONTESTCRASH undo_redo.dyn:103]: Yup, totally! Thanks again

**Pull request merged**